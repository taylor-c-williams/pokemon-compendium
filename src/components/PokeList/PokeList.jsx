import PropTypes from 'prop-types';
import PokeItem from '../PokeItem/PokeItem';

export default function PokeList({ pokedex }) {
	return (
		<div className="pokeList">
			{pokedex.map((pokemon) => {
				return <PokeItem {...pokemon} key={pokemon.id} />;
			})}
		</div>
	);
}

PokeList.propTypes = {
	pokedex: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.number.isRequired,
			name: PropTypes.string.isRequired,
			abilityOne: PropTypes.string.isRequired,
			abilityTwo: PropTypes.string.isRequired,
			typeOne: PropTypes.string.isRequired,
			typeTwo: PropTypes.string.isRequired,
			pic: PropTypes.string.isRequired,
			generation: PropTypes.string.isRequired,
		})
	).isRequired,
};
