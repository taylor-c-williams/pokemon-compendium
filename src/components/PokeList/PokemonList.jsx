import React from 'react';

export default function PokemonList(pokedex) {
	return (
		<div>
			{pokedex.map((pokemon) => (
				<h1 key={pokemon.pokemon}>{pokemon.pokemon}</h1>
			))}
		</div>
	);
}
