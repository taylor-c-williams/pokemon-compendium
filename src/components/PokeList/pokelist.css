.pokelist-body {
	display: grid;
	grid-template-columns: 1fr 1fr 1fr;
	justify-content: center;
	align-items: center;
	margin: 20px;
}
