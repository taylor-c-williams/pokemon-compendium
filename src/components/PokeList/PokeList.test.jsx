import { screen, render } from '@testing-library/react';
import PokeList from './PokeList';

it('accurately renders the PokeList component', () => {
	const pokemen = [
		{
			pokemon: 'bulbasaur',
			id: '1',
			type_1: 'grass',
			attack: '49',
			defense: '49',
			hp: '45',
			speed: '45',
			url_image:
				'http=//assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
		},
		{
			pokemon: 'ivysaur',
			id: '2',
			type_1: 'grass',
			attack: '62',
			defense: '63',
			hp: '60',
			speed: '60',
			url_image:
				'https://assets.pokemon.com/assets/cms2/img/pokedex/full/002.png',
		},
	];
	render(<PokeList pokedex={pokemen} />);
	const pokedex = screen.getAllByLabelText('pokemons!');
	expect(pokedex).toMatchSnapshot();
});
