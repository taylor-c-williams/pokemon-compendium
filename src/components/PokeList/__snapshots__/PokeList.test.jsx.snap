// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accurately renders the PokeList component 1`] = `
Array [
  <div
    aria-label="pokemons!"
    class="pokelist-body"
  >
    <section
      class="pokeitem-body"
    >
      <img
        alt="bulbasaur"
        src="http=//assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
      />
      <h1>
         
        bulbasaur
         
      </h1>
      Pokedex Number: 
      1
      Type: 
      grass
      HP: 
      45
      ATK: 
      49
      DEF: 
      49
      SPD: 
      45
    </section>
    <section
      class="pokeitem-body"
    >
      <img
        alt="ivysaur"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/002.png"
      />
      <h1>
         
        ivysaur
         
      </h1>
      Pokedex Number: 
      2
      Type: 
      grass
      HP: 
      60
      ATK: 
      62
      DEF: 
      63
      SPD: 
      60
    </section>
    ;
  </div>,
]
`;
